crates/bevy_ecs/src/world/mod.rs

```diff
@@ -469,6 +511,49 @@ impl World {
         }
     }
 
+    /// Gets mutable access to multiple entities at once, whose number is determined at runtime.
+    ///
+    /// # Panics
+    ///
+    /// If any entities do not exist in the world,
+    /// or if the same entity is specified multiple times.
+    ///
+    /// # Examples
+    ///
+    /// Disjoint mutable access.
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// # let mut world = World::new();
+    /// # let id1 = world.spawn_empty().id();
+    /// # let id2 = world.spawn_empty().id();
+    /// // Disjoint mutable access.
+    /// let mut entities = world.get_many_entities_dynamic_mut(&[id1, id2]).unwrap();
+    /// let entity1 = entities.get_mut(0).unwrap();
+    /// ```
+    ///
+    /// Trying to access the same entity multiple times will fail.
+    ///
+    /// ```should_panic
+    /// # use bevy_ecs::prelude::*;
+    /// # let mut world = World::new();
+    /// # let id = world.spawn_empty().id();
+    /// world.many_entities_dynamic_mut(&[id, id]);
+    /// ```
+    pub fn many_entities_dynamic_mut<'w>(&'w mut self, entities: &[Entity]) -> Vec<EntityMut<'w>> {
```

Comment:
I don't think we should include the panicking variants here. They're not actually any faster, and this isn't common enough to be a serious ergonomics issue.